### ISSUE TITLE
Provide better errors for failures to clear the apt cache

### ISSUE THREAD

**Issue by mr-cal** [canonical/snapcraft#4558]
### What needs to get done

The apt cache may fail to be removed when snapcraft is run in destructive-mode as root, then not as root.

Snapcraft should provide a better error for when the apt-cache cannot be removed.
This can be done by wrapping [this block](https://github.com/snapcore/snapcraft/blob/89961cfc3a9f93b3fe551711e5a4f1e4f0ef3fa8/snapcraft_legacy/internal/repo/apt_cache.py#L120) with a try/except and providing a useful and actionable error message.

You can reproduce this with:
```bash
lxc launch ubuntu:22.04 repro-env
lxc exec repro-env bash
apt update
apt install aptdaemon -y # to install /etc/apt/apt.conf.d/20dbus
snap install snapcraft --classic
su ubuntu
cd ~
mkdir test-snap
cd test-snap
snapcraft init
sed -i 's/core22/core20/g' snap/snapcraft.yaml
echo "    stage-packages: [curl]" >> snap/snapcraft.yaml
snapcraft --destructive-mode
snapcraft clean --destructive-mode
sudo --preserve-env=HOME snapcraft --destructive-mode  # --preserve-env=HOME not needed in ubuntu<19.10
sudo --preserve-env=HOME snapcraft clean --destructive-mode
snapcraft --destructive-mode
```

Error:
```
Sorry, an error occurred in Snapcraft:
[Errno 13] Permission denied: 'auth.conf.d'
We would appreciate it if you anonymously reported this issue.
No other data than the traceback and the version of snapcraft in use will be sent.
```

### Why it needs to get done

If snapcraft is run in destructive mode with sudo then without sudo, the error message doesn't help the user figure out how to clean their cache.

**Comment by lupaulus** [comment]
Hey, how do I clean the cache in Sudo, to get back in the normal modus

**Comment by mr-cal** [comment]
@lupaulus Try `sudo rm -r ~/.cache/snapcraft/stage-packages/`

**Comment by mr-cal** [comment]
You might have to delete the `~/.cache/snapcraft` dir.

**Comment by lupaulus** [comment]
> You might have to delete the `~/.cache/snapcraft` dir.

Yeah, back on track ! Don't forget to remove the root generated `.snap`, `prime` and `stage `folders

**Comment by mr-cal** [comment]
I'm closing this issue because we made a related improvement for snapcraft 8.14. Now, snapcraft warns when running in destructive-mode as a non-root user (https://github.com/canonical/craft-application/pull/974).
